src/core/utils/init_db.c

```diff
@@ -0,0 +1,98 @@
+#include "leveldb/c.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include "string.h"
+#include "init_db.h"
+
+int set_db_path(char **dest, char *name){
+  if(!getenv("HOME")){
+    exit(1);
+  }
+  char* home_path = getenv("HOME");
+  *dest = malloc(strlen(home_path)+strlen(name)+1);
+  strcpy(*dest, home_path);
+  strcat(*dest, name);
+  return 1;
+}
+
+int open_or_create_db(leveldb_t **db, char *path){
+  if(*db != NULL){
+    fprintf(stderr, "Database Already Open\n");
+    return 0;
+  }
+  leveldb_options_t *options;
+  char *err = NULL;
+  int ret = 0;
+  options = leveldb_options_create();
+  leveldb_options_set_create_if_missing(options, 1);
+  leveldb_options_set_compression(options, 0);  // Turn Off compression
+  //leveldb_options_set_cache(options, leveldb_cache_create_lru(100 * 1048576)); // Set large cache size@
+  *db = leveldb_open(options, path, &err);
+  if (err != NULL) {
+    fprintf(stderr, "Open fail: %s\n", err);
+    ret = 1;
+  }
+  /* reset error var */
+  leveldb_options_destroy(options);
+  leveldb_free(err);
+  return ret;
+}
+
+int init_db(leveldb_t **db, char **dest, char *name){
+  set_db_path(dest, name);
+  if(open_or_create_db(db, *dest) != 0){
+    return 1;
+  }
+  return 0;
+}
+
+int check_if_db_loaded(leveldb_t **db, char *path){
```

Comment:
See above comment, don't think this function really works in practice. Let's just make sure we init globals (should only have to happen once at the start anyways).